setup.cfg

```diff
@@ -53,7 +55,7 @@ install_requires =
     tweakwcs>=0.8.3
     asdf-astropy>=0.3.0
     wiimatch>=0.2.1
-    packaging>19.0
+    packaging>20.0
```

Comment:
This was required for the minimum version (0.19) of scikit-image.

Reply:
Looks ok to me. From our slack messages I assume you have run the outlier detection regression tests and they now show you no difference

Reply:
Thanks for taking a look.

 I should clarify, this PR does cause 12 regtest failures. The failures are from 2 parameterized tests:
- [test_spec3_multi_emsm](https://github.com/spacetelescope/jwst/blob/2199ba88fe0a74f825733dc7114a3f6c79c581c7/jwst/regtest/test_nirspec_ifu_spec3_emsm.py#L40)
- [test_spec3_multi](https://github.com/spacetelescope/jwst/blob/master/jwst/regtest/test_nirspec_ifu_spec3.py#L50)

These are the same tests that fail when scipy is updated, or when run on a mac and can be explained by the undefined (and unreliable) output of `scipy.signal.medfilt` when provided a `np.nan` input (as occurs in the failing tests). To use the examples in the unit test added with this PR.

```python
arr = [2, np.nan, 0]
ksize = [3]
```

The expected output using a median filter (and filling edges with 0s as is done by default for `scipy.signal.medfilt`) should be `[1, 1, 0]`.

If I run the following on my mac, with scipy 1.9.3 I get a different and incorrect output:
```python
>>> scipy.signal.medfilt([2, np.nan, 0], [3])
array([nan, nan,  0.])
```
with scipy 1.11.3 I get a different (and also incorrect):
```python
>>> scipy.signal.medfilt([2, np.nan, 0], [3])
array([0., 0., 0.])
```
Other examples show differences for linux vs mac.

As the truth files were generated using `scipy.signal.medfilt` and the input contains `nan` the truth file contents depend on the undefined behavior of the version of scipy used to generate the file. The truth files for these tests will need to be updated if the output is acceptable (and this PR is merged).

Reply:
To further clarify, with this PR the new output (which differs from the truth files) is identical on mac and linux for the `_crf` and `_x1d` files. The `_s3d` files show differences (< 1e-5) on mac and linux (independent of this PR other tests that generate `_s3d` files show differences between mac and linux on the main branch as seen in the jenkins devdeps tests so this looks to be a separate issue, one not addressed in this PR).

For the `_crf` files, the differences are only for pixels with a `nan` within the 7x7 window used for the median filter. These are largely found on the edges of usable pixels. Taking the first example in the diff for file `jw01249005001_03101_00001_nrs1_o005_crf.fits`. The fitsdiff output reports:
```
Extension HDU 1 (SCI, 1):
   Data contains differences:
     Data differs at [1716, 94]:
        a> nan
        b> -103.99343
...
   Data contains differences:
     Data differs at [1716, 94]:
        a> 33554449
         ?       ^^
        b> 33554432
```
indicating that the pixel at (using numpy indices) [93, 1715] is being flagged as an outlier in the result file but not in the truth file. Inspecting the `minarr` calculated during the outlier detection step:
https://github.com/spacetelescope/jwst/blob/2199ba88fe0a74f825733dc7114a3f6c79c581c7/jwst/outlier_detection/outlier_detection_ifu.py#L220-L226
```python
>> minarr[93, 1715]
66.545156955719
>> minarr[90:97, 1712:1719]
array([[        nan,         nan,         nan,         nan,         nan,
                nan,         nan],
       [        nan,         nan,         nan,         nan,         nan,
                nan,         nan],
       [        nan,         nan,         nan,         nan,         nan,
                nan,         nan],
       [29.89870262,  4.97431731, 19.80752754, 66.54515696, 27.27770805,
        17.14254284,  2.87409228],
       [ 0.38395762,  2.3855598 ,  4.7994982 ,  8.4161818 ,  1.58876133,
         3.21722126,  1.05671763],
       [ 0.14560556,  1.19486213,  0.89728248,  0.40425563,  1.58876133,
         1.25159645,  0.71704447],
       [ 0.14560556,  1.19486213,  0.89728248,  0.40425563,  2.14912176,
         0.49365234,  0.51363105]])
```
This pixel is on the edge of a group of usable pixels, inspecting the computed `normarr` (the output of `medfilt`) for this PR we see:
```python
>>  normarr[90:97, 1712:1719]
array([[19.80752754, 27.27770805, 27.27770805, 19.80752754, 17.14254284,
        19.80752754, 17.14254284],
       [ 6.30404165,  7.02648562,  6.69524956,  4.88690776,  4.00835973,
         4.62222749,  4.19920981],
       [ 3.89869487,  3.89869487,  2.3855598 ,  2.3855598 ,  2.3855598 ,
         2.78641891,  2.78641891],
       [ 1.19486213,  1.58876133,  1.42017889,  1.42017889,  1.42017889,
         1.58876133,  1.58876133],
       [ 1.0041163 ,  1.58876133,  1.25159645,  1.25159645,  1.25159645,
         1.58876133,  1.51533907],
       [ 0.9249387 ,  1.30883139,  1.22322929,  1.22322929,  1.19486213,
         1.33719856,  1.46906986],
       [ 0.91056663,  1.42280066,  1.19486213,  1.25159645,  1.19486213,
         1.25159645,  1.42280066]])
```
Which differs substantially from the `normarr` computed without this PR (using `scipy.signal.medfilt`):
```python
>> scipy.signal.medfilt(minarr)[90:97, 1712:1719]
array([[        nan,         nan,         nan,         nan,         nan,
                nan,         nan],
       [        nan,         nan,         nan,         nan,         nan,
                nan,         nan],
       [19.80752754, 27.27770805, 27.27770805, 19.80752754,         nan,
        19.80752754, 17.14254284],
       [19.80752754,  8.83398247, 27.27770805, 19.80752754, 17.14254284,
        27.01134586,         nan],
       [ 3.89869487,  2.41227081,  2.41227081,  2.41227081,  2.2597177 ,
         2.78641891,  2.2597177 ],
       [ 1.19486213,  1.62179971,  1.58876133,  1.58876133,  1.58876133,
         1.62179971,  1.62179971],
       [ 0.91056663,  1.42280066,  1.19486213,  1.25159645,  1.19486213,
         1.25159645,  1.42280066]])
```
Inspecting the value of `normarr` used in the further outlier detection for the pixel in question gives 2 very different results for this PR vs main:
```python
>> scipy.signal.medfilt(minarr)[93, 1715]
19.807527542114258
>> normarr[93, 1715]
1.4201788902282715
```
As the `normarr` values are later used to compute the ratio of the minarr value relative to the local median this pixel with this PR is ~66 / 1.4 and flagged as an outlier vs ~66 / 19.8 and not flagged as an outlier. As this 'flag' is used to mark bad pixels in all 8 input files this explains the 8 `_crf` test failures.